pkg/cloudprovider/util/cloud_init_settings.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 The Machine Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"html/template"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CloudInitNamespace = "cloud-init-settings"
+	jwtTokenNamePrefix = "cloud-init-getter-token"
+
+	bootstrapCloudConfig = `#cloud-config
+
+write_files:
+- path: "/opt/bin/bootstrap"
+  permissions: "0755"
+  content: |
+    #!/bin/bash
+    set -xeuo pipefail
+    apt-get update
+    apt-get install jq -y
+    wget --no-check-certificate --quiet \
```

Comment:
We should probably retry here.

Reply:
Also, some recommendations:
- You can include the cluster cacert in this cloud-cloud init and use it instead of `--no-check-certificates`. You will need it later anyway
- You can either use `curl` or use `wget -O -` to right to stdout directly. and avoid the temp file.
- curl/wget might exist on the node based on the image. So might be a good idea to apt install it anyway.